plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -110,12 +113,7 @@ public CompletionStage<PutDataObjectResponse> putDataObjectAsync(PutDataObjectRe
                 final PutItemRequest putItemRequest = PutItemRequest.builder().tableName(tableName).item(item).build();
 
                 dynamoDbClient.putItem(putItemRequest);
-                String simulatedIndexResponse = simulateOpenSearchResponse(
-                    request.index(),
-                    request.id(),
-                    source,
-                    Map.of("result", "created")
-                );
+                String simulatedIndexResponse = simulateOpenSearchResponse(request.index(), id, source, Map.of("result", "created"));
```

Comment:
This needs SeqNo incremented and returned in the simulatedIndexResponse.

Reply:
Concurrent support in DDB client is not in scope for this PR.